### PR TITLE
[Backport 2.x] Bump org.apache.maven:maven-model from 3.6.2 to 3.9.1

### PR DESCRIPTION
**Pull request by owaiskazi19**
### Description
Backport of https://github.com/opensearch-project/OpenSearch/pull/6753

### Issues Resolved
[List any issues this PR will resolve]

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff
- [ ] Commit changes are listed out in CHANGELOG.md file (See: [Changelog](../blob/main/CONTRIBUTING.md#changelog))

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by owaiskazi19** [comment]
> ### Gradle Check (Jenkins) Run Completed with:
> 
>     * **RESULT:** FAILURE ❌
> 
>     * **URL:** https://build.ci.opensearch.org/job/gradle-check/12721/
> 
>     * **CommitID:** [a3b954e](https://github.com/opensearch-project/OpenSearch/commit/a3b954ea91284854b2711e090107069d9aca39e1)
>       Please examine the workflow log, locate, and copy-paste the failure(s) below, then iterate to green.
>       Is the failure [a flaky test](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md#flaky-tests) unrelated to your change?

https://github.com/opensearch-project/OpenSearch/issues/5958

File: CHANGELOG.md
Comment (reta) [inline CHANGELOG.md:27]: That seems to be taking changes from other backports?

Comment (owaiskazi19) [reply CHANGELOG.md:27]: I added this to avoid conflicts and pushing again. That's why added in the PR description to merge this PR after #6768 is merged.

Comment (owaiskazi19) [reply CHANGELOG.md:27]: Got the conflict eventually. Updated it

**Review by reta — APPROVED** [review]

**Comment by owaiskazi19** [comment]
> ### Gradle Check (Jenkins) Run Completed with:
> 
>     * **RESULT:** FAILURE ❌
> 
>     * **URL:** https://build.ci.opensearch.org/job/gradle-check/12726/
> 
>     * **CommitID:** [494f2ce](https://github.com/opensearch-project/OpenSearch/commit/494f2ce373efc4e36e4527d140fbf5e22b8da38d)
>       Please examine the workflow log, locate, and copy-paste the failure(s) below, then iterate to green.
>       Is the failure [a flaky test](https://github.com/opensearch-project/OpenSearch/blob/main/DEVELOPER_GUIDE.md#flaky-tests) unrelated to your change?

https://github.com/opensearch-project/OpenSearch/issues/6738

**Pull request merged**
